realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -516,8 +547,18 @@ private Row checkRealmObjectConstraints(String fieldName, @Nullable RealmModel v
                 checkTypeOfListElements(list, Double.class);
                 osResults.setDoubleList(fieldName, (RealmList<Double>) list);
                 break;
-            default:
+            default: {
+                // Handle Decimal128 and ObjectId in a special way since they might not be on the
+                // classpath
+                if (columnType == RealmFieldType.DECIMAL128_LIST) {
```

Comment:
As discussed, let's add the bson dependency for this PR and optimise later a mechanism to make it optional. I think the method count and size will not affect apps not using these types (because of R8 tree shaking)